presentation/src/main/java/me/androidbox/presentation/event/viewmodel/EventViewModel.kt

```diff
@@ -4,18 +4,34 @@ import android.util.Log
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
 import kotlinx.coroutines.launch
 import me.androidbox.domain.authentication.ResponseState
+import me.androidbox.domain.authentication.model.Event
 import me.androidbox.domain.authentication.remote.EventRepository
-import java.util.UUID
+import me.androidbox.presentation.event.screen.EventScreenEvent
+import me.androidbox.presentation.event.screen.EventScreenState
+import java.util.*
 import javax.inject.Inject
-import me.androidbox.domain.authentication.model.Event
 
 @HiltViewModel
 class EventViewModel @Inject constructor(
-    private val eventRepository: EventRepository
+    private val eventRepository: EventRepository,
 ) : ViewModel() {
 
+    private val _eventScreenState: MutableStateFlow<EventScreenState> = MutableStateFlow(EventScreenState())
+    val eventScreenState = _eventScreenState.asStateFlow()
+
+    fun onEventScreenEvent(eventScreenEvent: EventScreenEvent) {
```

Comment:
Lets say you add more events

You will have one big viewmodel with tons of events that arent related to a single screen

I personally prefer having 1 vm per screen
So everything from the vm is related to that exact screen

Reply:
@MKiperszmid this component will be added to the event screen. This viewmodel is for the event screen.

Reply:
Yes I feel like @MKiperszmid misunderstands this event screen, this looks very fine to me